modules/carousel/jetpack-carousel.php

```diff
@@ -545,13 +548,15 @@ function get_attachment_comments() {
 	function post_attachment_comment() {
 		if ( ! headers_sent() )
 			header('Content-type: text/javascript');
+		
+		$post_nonce = filter_input( INPUT_POST, 'nonce' );
 
-		if ( empty( $_POST['nonce'] ) || ! wp_verify_nonce($_POST['nonce'], 'carousel_nonce') )
+		if ( empty( $post_nonce ) || ! wp_verify_nonce($post_nonce, 'carousel_nonce') )
 			die( json_encode( array( 'error' => __( 'Nonce verification failed.', 'jetpack' ) ) ) );
 
-		$_blog_id = (int) $_POST['blog_id'];
-		$_post_id = (int) $_POST['id'];
-		$comment = $_POST['comment'];
+		$_blog_id	 = filter_input( INPUT_POST, 'blog_id', FILTER_VALIDATE_INT );
+		$_post_id	 = filter_input( INPUT_POST, 'id', FILTER_VALIDATE_INT );
+		$comment	 = filter_input( INPUT_POST, 'comment', FILTER_VALIDATE_INT );
```

Comment:
Comment is not an integer, I'd rather not use any validation/sanitization on this string to avoid unneeded changes.